ch_backup/logging.py

```diff
@@ -2,67 +2,151 @@
 Logging module.
 """
 
+import inspect
 import logging
-import logging.config
-import os
+from typing import Any
 
 import psutil
+from loguru import logger
 
 from ch_backup.util import format_size
 
 
-def configure(config: dict) -> None:
+class Filter:
     """
-    Configure logging.
+    Filter for luguru handler.
     """
-    for handler in config.get("handlers", {}).values():
-        filename = handler.get("filename")
-        if filename:
-            os.makedirs(os.path.dirname(filename), exist_ok=True)
 
-    logging.config.dictConfig(config)
+    def __init__(self, name):
+        self._name = name
+
+    def __call__(self, record):
+        """
+        Filter callback to decide for each logged message whether it should be sent to the sink or not.
+
+        If the log comes from ch-backup code then the logger name will be in `logger_name`.
+        If the log comes from other module and it caught by InterceptHandler then we filtering by the module_name.
+        """
+
+        if "logger_name" in record.get("extra", {}):
+            return record["extra"].get("logger_name") == self._name
+
+        if "module_name" not in record.get("extra", {}):
+            return False
+
+        if record["extra"]["module_name"][: len(self._name)] == self._name:
+            record["extra"]["logger_name"] = self._name
+            return True
+        return False
+
+
+def make_filter(name):
+    """
+    Factory for filter creation.
+    """
+
+    return Filter(name)
+
+
+class InterceptHandler(logging.Handler):
+    """
+    Helper class for logging interception.
+    """
+
+    def emit(self, record: logging.LogRecord) -> None:
+        """
+        Intercept all records from the logging module and redirect them into loguru.
+
+        The handler for loguru will be chosen based on module name.
+        """
+
+        # Get corresponding Loguru level if it exists.
+        level: int or str  # type: ignore
+        try:
+            level = logger.level(record.levelname).name
+        except ValueError:
+            level = record.levelno
+
+        # Find caller from where originated the logged message.
+        frame, depth = inspect.currentframe(), 0
+        while frame and (depth == 0 or frame.f_code.co_filename == logging.__file__):
+            frame = frame.f_back
+            depth += 1
+
+        try:
```

Comment:
Let's remove this and put `bind(logger_name=record.name)` to pass original logger name to the handlers